### PR TITLE
Add contextual help to Analysis UI

### PR DESCRIPTION
**Pull request by Jesus89**
Related to: #11907

Changes
- Add "MORE INFO" link in analysis option modals [1]
  - \+ `track-[analysis type]-modal-learn` measurement class
- Add "MORE INFO" link in analysis forms [1]
  - \+ `track-[analysis type]-form-learn` measurement class
- Remove info-analysis-panel
- Remove unused code

analysis-links

:star2: Bonus
- Fix Analysis model styles
  - Add missing border
  - Darker texts
  - Fix padding
  - Bigger "Add analysis" button

| Before | After |
|---|---|
|analysis-modal-before|analysis-modal-after|

For acceptance:
1. Open Analysis modal
2. Click any "MORE INFO" link
3. Create an analysis
4. In the form click the "MORE INFO" link

**[1]** Links are specified in the related issue #11907.

```
* Enrich from Data Observatory: data-observatory-measure
                                data-observatory-multiple-measures
* Filter by Layer: filter-by-node-column
* Georeference: georeference-long-lat
* Join Columns from 2nd Layer: merge
* Intersect Second Layer: aggregate-intersection
* Create Areas of Influence: buffer
* Find Centroid of Geometries: centroid
* Group points into polygons: convex-hull
* Filter by Column Value: filter-range
* Filter points in polygons: intersection
* Connect with Lines: line-to-single-point
* Subsample percent of rows: sampling
* Calculate Clusters of Points: kmeans
* Detect Outliers and Clusters: moran
* Predict Trends and Volatility: spatial-markov-trend
* Find nearest: closest
```

### CODE REVIEW

File: app/assets/stylesheets/editor-3/_header.scss [hunk 1]
```diff
@@ -76,6 +76,10 @@
 .Editor-HeaderInfo-actionsItem:first-child {
   margin-left: 0;
 }
+.Editor-HeaderInfo-link {
+  margin-right: 4px;
+  font-weight: bold;
```
Comment (nobuti) [inline app/assets/stylesheets/editor-3/_header.scss]: we should use the weights already available in cartoassets: https://github.com/CartoDB/CartoAssets/blob/master/src/scss/cdb-variables/_sizes.scss#L23-L25

File: app/assets/stylesheets/editor-3/_modals-elements.scss [hunk 2]
```diff
@@ -78,6 +78,16 @@
...
   border-radius: $halfBaseSize;
 }
+
+.Analysis-info {
+  padding: 16px;
+}
+
+.Analysis-link {
+  display: inline-flex;
+  font-weight: bold;
```
Comment (nobuti) [inline app/assets/stylesheets/editor-3/_modals-elements.scss]: ⬆️

File: app/assets/stylesheets/editor-3/_header.scss [hunk 3]
```diff
@@ -76,6 +76,10 @@
 .Editor-HeaderInfo-actionsItem:first-child {
   margin-left: 0;
 }
+.Editor-HeaderInfo-link {
+  margin-right: 4px;
```
Comment (nobuti) [inline app/assets/stylesheets/editor-3/_header.scss]: we have an utility for this too https://github.com/CartoDB/CartoAssets/blob/master/src/scss/cdb-utilities/_helpers.scss#L17-L19

File: lib/assets/core/javascripts/cartodb3/components/modals/add-analysis/analysis-option-view.js [hunk 4]
```diff
@@ -66,12 +66,9 @@ module.exports = CoreView.extend({
...
 
-  _onClick: function (e) {
-    if (e && e.target && $(e.target).hasClass('js-more')) {
-      this.trigger('onClickInfo', this);
-    }
-
-    if (this._acceptsInputGeometry()) {
+  _onClick: function (event) {
+    var linkClicked = event && event.target && $(event.target).hasClass('js-more');
+    if (!linkClicked && this._acceptsInputGeometry()) {
```
Comment (nobuti) [inline lib/assets/core/javascripts/cartodb3/components/modals/add-analysis/analysis-option-view.js]: Sweet!

File: lib/assets/core/javascripts/cartodb3/data/analyses.js [hunk 5]
```diff
@@ -17,6 +17,7 @@ var MAP = {
     modalTitle: _t('components.modals.add-analysis.option-types.aggregate-intersection.title'),
     modalDesc: _t('components.modals.add-analysis.option-types.aggregate-intersection.desc'),
     modalCategory: 'data_transformation',
+    modalLink: 'https://carto.com/learn/guides/analysis/intersect-second-layer',
```
Comment (nobuti) [inline lib/assets/core/javascripts/cartodb3/data/analyses.js]: Not sure if we should localized this links, are they only existing in english?

File: lib/assets/core/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-view.js [hunk 6]
```diff
@@ -40,7 +43,13 @@ module.exports = CoreView.extend({
       model: this._formModel,
       trackingClass: 'track-' + this._formModel.get('type') + '-analysis',
       template: this._formModel.getTemplate(),
-      templateData: this._formModel.getTemplateData()
+      templateData: _.extend(
+        this._formModel.getTemplateData(),
+        {linkContent: linkTemplate({
```
Comment (nobuti) [inline lib/assets/core/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-view.js]: React! Just kidding, let's put a new line opening and closing the object please.

**Comment by arianaescobar** [comment]
I can't review it because it won't load the analysis modal :( 

It get's stuck here:

screen shot 2017-09-26 at 12 52 09

**Comment by Jesus89** [comment]
It is not deployed yet. I will notify you when its done.

**Comment by arianaescobar** [comment]
Nevermind, sorted. 

It loaded, but I can't see anything about the contextual help in the analysis UI. I'm in a ded15 user, can you confirm it has been correctly deployed there?

**Comment by arianaescobar** [comment]
oh! haha ok! :)

**Comment by Jesus89** [comment]
Deployed (in ded15). Try again!

**Comment by arianaescobar** [comment]
Ahora sí! 

🇻🇪 

Just two things... is this normal? 

screen shot 2017-09-26 at 14 07 57
screen shot 2017-09-26 at 14 08 12

And, is it possible there's an extra space after "MORE INFO" in the Analysis tab UI? 

Analysis tab
screen shot 2017-09-26 at 14 11 26

Analysis modal
screen shot 2017-09-26 at 14 12 31

**Comment by nobuti** [comment]
Yes, all users in staging has the autogenerated analysis feature flag.

**Comment by Jesus89** [comment]
> And, is it possible there's an extra space after "MORE INFO" in the Analysis tab UI?

Yes, of course. Please, add the top/right padding values. Thanks!

**Comment by arianaescobar** [comment]
Sorry, I think I explained that poorly. I meant that there is (or looks like it) an extra space after the O in MORE INFO that we should remove.

**Comment by Jesus89** [comment]
Good point! Let me check that.

**Comment by Jesus89** [comment]
@arianaescobar Deployed with the changes in ded15

**Comment by arianaescobar** [comment]
B A N D E R A

**Pull request merged**